source/MaterialXGenShader/SgNode.cpp

```diff
@@ -355,6 +378,18 @@ void SgNode::renameOutput(const string& name, const string& newName)
     }
 }
 
+bool SgNode::elementCanBeSampled2D(const Element& element) const
```

Comment:
Default what can be sampled, can extend later on if we have otherways to determine this.